features/todo/todoService.go

```diff
@@ -17,7 +20,7 @@ func NewTodoService() *TodoService {
 	s := storage.StorageAdapterFactory{}
 	storageAdapter, err := s.GetInstance(storage.DEFAULT)
 	if err != nil {
-		log.Fatalf("failed to create TodoService instance: %s", err.Error())
```

Comment:
Please consider when replacing `Fatalf` with higher log levels, what happens if storage adapter can't be instantiated ?